vizro-core/tests/unit/vizro/models/test_types.py

```diff
@@ -8,113 +8,111 @@
 from vizro.models.types import CapturedCallable, capture
 
 
-@pytest.fixture
-def varargs_function():
-    def function(*args, b=2):
-        return args[0] + b
-
-    return CapturedCallable(function, 1)
-
+def positional_only_function(a, /):
+    pass
 
-@pytest.mark.xfail
-# Known bug: *args doesn't work properly. Fix while keeping the more important test_varkwargs
-# passing due to https://bugs.python.org/issue41745.
-# Error raised is IndexError: tuple index out of range
-def test_varargs(varargs_function):
-    assert varargs_function(b=2) == 1 + 2
 
+def var_positional_function(*args):
+    pass
 
-@pytest.fixture
-def positional_only_function():
-    def function(a, /, b):
-        return a + b
-
-    return CapturedCallable(function, 1)
 
+@pytest.mark.parametrize("function", [positional_only_function, var_positional_function])
+def test_invalid_parameter_kind(function):
+    with pytest.raises(
+        ValueError,
+        match="CapturedCallable does not accept functions with positional-only or variadic positional parameters",
+    ):
+        CapturedCallable(function)
 
-@pytest.mark.xfail
-# Known bug: position-only argument doesn't work properly. Fix while keeping the more important
-# test_varkwargs passing due to https://bugs.python.org/issue41745.
-# Error raised is TypeError: function got some positional-only arguments passed as keyword arguments: 'a'
-def test_positional_only(positional_only_function):
-    assert positional_only_function(b=2) == 1 + 2
 
+def positional_or_keyword_function(a, b, c):
+    return a + b + c
 
-@pytest.fixture
-def keyword_only_function():
-    def function(a, *, b):
-        return a + b
 
-    return CapturedCallable(function, 1)
+def keyword_only_function(a, *, b, c):
+    return a + b + c
 
 
-def test_keyword_only(keyword_only_function):
-    assert keyword_only_function(b=2) == 1 + 2
+def var_keyword_function(a, **kwargs):
+    return a + kwargs["b"] + kwargs["c"]
 
 
 @pytest.fixture
-def varkwargs_function():
-    def function(a, b=2, **kwargs):
-        return a + b + kwargs["c"]
-
-    return CapturedCallable(function, 1)
-
-
-def test_varkwargs(varkwargs_function):
-    varkwargs_function(c=3, d=4) == 1 + 2 + 3
-
-
-@pytest.fixture
-def simple_function():
```

Comment:
`simple_function` (now called `positional_or_keyword_function`) is tested on the same suite as before, just with order changed a bit to facilitate parametrisation of tests.